### PR TITLE
[WIP] vmcloak: fix PKGBUILD

### PR DESCRIPTION
**Pull request by noraj**
see #2474
WIP because install still failing

### CODE REVIEW

**Comment by Edu4rdSHL** [comment]
@noraj you need to patch `setup.py` and remove the static versions required here: https://github.com/hatching/vmcloak/blob/master/setup.py#L42-L58. They should look:

```
install_requires=[
        "click",
        "jinja2",
        "pefile2",
        "pyyam",
        "sqlalchemy",
        "alembic",
    ],
...
```

**Comment by noraj** [comment]
It shouldn't mater as we are installing dependencies in a virtual environment `python setup.py install --root="$pkgdir" --prefix=/usr` so they can bring exact version if they want to. Removing the version requirement will work but I wonder if some are not respected they could break the app.

**Comment by Edu4rdSHL** [comment]
It matter in all aspects. In the PKGBUILD we put the dependency name and it fetch the latest version in our repos.

When you install using the setup.py file with explicit versions numbers it make the version of the dependency a condition, that's why putting specific versions there make sense. If you look in out python packages, there are a lot of patches just to remove that thing.

**Comment by noraj** [comment]
I don't understand what you mean. All dependencies listed in setup.py are not in `PKGBUILD`. `python setup.py install --root="$pkgdir" --prefix=/usr` will fetch the exact version required for the package and install them in a virtual folder out of the system python tree. So this don't fetch packages from our repo but from pypi nad we use a virtual folder exactly for not caring about dependencies requirements not being updated. If teh app use the dependencies install in the virtual environment which were exactly the ones required in setup.py I don't understand how it can fails. Please explain me on discord if I didn't understand something.

**Comment by Edu4rdSHL** [comment]
@noraj please can you just patch the setup file and try again? You can check it yourself.

TL;DR:

When you build a Python package using specific versions it try to use the specific versions when installed, if you don't use specific versions in setup.py it try to work with any version. Python packages using setup.py doesn't download anything using pip, they use system packages (that's why they are included in depends=().

**Comment by noraj** [comment]
I pushed the fix but the problem is that the BA package [python2-pefile] seems to match pypi [pefile](https://pypi.org/project/pefile/) but setup.py requires pypi [pefile2](https://pypi.org/project/pefile2/) instead are form the same repo only old releases like 1.2.11 (before 2013) after that date the releases changed to year.month.day format.
Since it isn't a different package but only a old version of an existing one I'll open an upstream bug.

**Comment by noraj** [comment]
@Edu4rdSHL as all dependencies are very outdated anyway and upstream won't upgrade anytime soon we may need to switch to a [pip](https://wiki.archlinux.org/index.php/Python_package_guidelines#pip) install as we are already doing with gem for all ruby packages.

**Comment by Edu4rdSHL** [comment]
We should remove the package instead or wait for upstream changes, using pip to install things as root is one of the worse things I've seen and will only cause most problems that solving. 

What happens if you change the dependency from pefile2 to pefile instead? Or well, we can package a python2-pefile2 package as well.

**Comment by noraj** [comment]
I said **all the dependencies are very outdated**. They all have one or several MAJOR version late with several breaking changes not only for pefile but for all of them. So there is no way it will work with latest version of all of them.

For  ruby we install all the dependencies in a virtual vendor folder under `/usr/share/$pkgname/`, I think it is possible to do the same with python. I never thought of using pip to install packages globally with the system ones.

**Comment by noraj** [comment]
Something like

```
pip install -r requirements.txt -t vendor
```

As it's done for composer or npm by default or for ruby with `bundle install --path vendor/bundle`

https://stackoverflow.com/questions/52695451/install-python-package-at-current-directory

**Comment by mrsmith0x00** [comment]
@noraj is it done for merge?

**Comment by noraj** [comment]
@mrsmith0x00 no I gave up

**Comment by noptrix** [comment]
hey guys. i fixed this package and did a version bump (switched to git master branch, vcs version): https://github.com/BlackArch/blackarch/commit/d4447a52431d903ba3a40edb7a57256b69f0b273
